File: frontend/server/payments/gateway.js
import { MockGateway } from './drivers/mock.js';
import { StripeGateway } from './drivers/stripe.js';
import { RazorpayGateway } from './drivers/razorpay.js';

export function getGatewayDriver(provider = process.env.PAYMENT_GATEWAY_PROVIDER || 'mock') {
  switch (provider.toLowerCase()) {
    case 'stripe':
      return new StripeGateway();
    case 'razorpay':
      return new RazorpayGateway();
    case 'mock':
    default:
      return new MockGateway();
  }
}

File: backend/src/services/payments/gateway.py
from src.core.config import settings
from src.services.payments.drivers import MockGateway, StripeGateway, RazorpayGateway

def get_gateway_driver(provider: str = None):
    if not provider:
        provider = settings.PAYMENT_GATEWAY_PROVIDER or "razorpay"
        
    p = provider.lower()
    if p == "mock" and settings.ENVIRONMENT.lower() == "production":
        raise RuntimeError("Mock payment gateway is disabled in production")
    if p == "stripe":
        return StripeGateway()
    elif p == "razorpay":
        return RazorpayGateway()
    else:
        return MockGateway()